saleor/plugins/vatlayer/__init__.py

```diff
@@ -89,3 +98,131 @@ def get_taxed_shipping_price(shipping_price, taxes):
     if not charge_taxes_on_shipping():
         taxes = None
     return apply_tax_to_price(taxes, DEFAULT_TAX_RATE_NAME, shipping_price)
+
+
+def apply_checkout_discount_on_checkout_line(
+    checkout_info: "CheckoutInfo",
+    lines: List["CheckoutLineInfo"],
+    checkout_line_info: "CheckoutLineInfo",
+    discounts: Iterable["DiscountInfo"],
+    line_price: Money,
+):
+    """Calculate the checkout line price with discounts.
+
+    Include the entire order voucher discount.
+    """
+    voucher = checkout_info.voucher
+    if not voucher or voucher.type == VoucherType.SHIPPING:
+        return line_price
+
+    line_quantity = checkout_line_info.line.quantity
+    total_discount_amount = checkout_info.checkout.discount_amount
+    line_total_price = line_price * line_quantity
+    currency = checkout_info.checkout.currency
+
+    # if the checkout has only one line, the hole discount amount will be applied
+    # to this line
+    if len(lines) == 1:
+        return (
+            line_total_price - Money(total_discount_amount, currency)
+        ) / line_quantity
+
+    # if the checkout has more lines we need to propagate the discount amount
+    # proportionally to total prices of items
+    lines_total_prices = [
```

Comment:
Maybe I am missing something, but would be faster and easier to define the discount that should be applied for single value of given currency (like 1 USD, or 1PLN)?
Something like this:
  ```
  line_1 = Decimal(10)
  line_2 = Decimal(80)
  subtotal = line_1 + line_2
  discount_percent_value = 0.5 # 50%
  discount = subtotal * discount_percent_value
  discount_per_single_unit = discount/subtotal # 45/90 == 0.5 USD per 1USD
  discount_line_1 = line_1 * discount_per_single_unit
  discount_line_2 = line_2 * discount_per_single_unit
  ```
  I assume that maybe I am missing something but thought that it would be good to mention